### DISCUSSION TITLE
Proton?

### DISCUSSION THREAD

**Discussion by duckcantcode** [danoon2/Boxedwine#88] [Feature Requests]
Is it possible we could use Proton with Boxedwine? Proton is known to work with many games, including the ones I've had trouble getting working in Boxedwine, but we would have to build it for 32-bit Linux somehow. If Proton worked in Boxedwine that would unlock compatibility with hundreds of more games.

For what it's worth, I'm currently trying to build Proton under 32-bit Debian, but I'm not sure it's going to work.

**Comment by danoon2** [comment]
Proton does sound interesting, at a certain point I assume its just patches for Wine.  Currently I support up to about Wine 8.0 (only tested up to Wine 7.0).  Nothing after that will work because I completely replaced winex11.drv.so with my own that allows me to handle drawing to the screen instead of needing to emulated x11.  Every time they modify the GDI structure in the Wine source, I need to make a change in Boxedwine.  See

https://github.com/danoon2/Boxedwine/blob/963b6e524274c35ee0ff28126c825db7ee47b8a6/tools/wineboxed.drv/driver.c#L3380

Wine 6 and 7 saw a lot of changes.  The biggest one is now they used shared memory to communicate things like tick count between wineserver and the clients.  That caused a lot new of new code in Boxedwine and the cpu emulation.

I currently build my version of winex11.drv when I build Wine.  see

https://github.com/danoon2/Boxedwine/wiki/How-build-Wine

If you can get Wine 6 to build with my how too, you should be able to apply patches to it.

**Reply by duckcantcode** [reply]
I'm not too worried about building the rootfs, it seems like you streamlined the process (thank you!). The issue is getting Proton to build for a 32-bit system.

Part of it is patches for wine but it seems like there's other stuff added like dxvk and Vulkan stuff, as well as OpenXR etc. Check out Proton here https://github.com/ValveSoftware/Proton and let me know if it seems doable.
